### ISSUE TITLE
Cool, Linux can run executables, but now Windows can't

### ISSUE THREAD

**Issue by StuffOfSonny** [FizzleFuze/SteamNTFS#1]
So to preface, I was not, and still am not fully ready to swap from Windows to Linux. I had intended/still intend to take my time, but now I'm faced with the issue of Windows no longer run its own "proprietary" files anymore. I get a "Access is Denied" error when I try to run Steam/its games. I've tried going into Properties -> Security, and adding Full Control to various principals, and it only works until a file is updated (Game patch, Steam Update, Windows Update, **New Game Install**). Is there any way I can essentially do this in reverse, to unbreak my Steam Folder?

**Comment by FizzleFuze** [comment]
StuffOfSonny,

This sounds very frustrating, sorry it's giving you grief. This is a very interesting situation to me though, so I'd like to learn more about it. 

The only suggestion I can give you off the top of my head, is to change the owner and group back to root, then unmount (remove from fstab to stop future mounts) on the Linux side. If you would like further suggestions, I'll need more info from you. 

I need at least to know which Linux distro and version you're using and which version of Windows, as well as all the details you can give me about the security policies and configuration on the Windows side.

Anything from the logs on either system would be helpful as well.

The more you tell me, the better I can help. :)

**Comment by StuffOfSonny** [comment]
I decided to try out Pop!_OS as my Linux distro. I think the distro updated recently, and if it has, I haven't updated my install yet. As for the Windows side of things, I'm on Windows 10 21H1

As for permissions examples, I'm assuming this "unknown account" is Linux, and everything in my Steam Folder specifically looks like this (I had to use a specific game that I haven't done the temporary fix yet to.
![explorer_2021-07-15_17-23-57](https://user-images.githubusercontent.com/61560546/125859908-c943746e-64e2-415d-bcca-2e910cfd2ec7.png)

And this is what it looks like on the Administrator's account
![explorer_2021-07-15_17-27-17](https://user-images.githubusercontent.com/61560546/125860005-d1b64605-7f8a-4113-aac0-9da525caabca.png)

No matter what I seem to do, that "full control" permission gets unticked. I think it does this every time files relating to it update,  (so patches and Steam/Windows updates), but some games have played nicely for some reason. Sonic Mania is one I've only needed to fix once, but my guess is that it has to do with Steam DRM.

**Comment by FizzleFuze** [comment]
Thank you for the detailed information. 

I am particularly interested in your findings that "full control" permission gets removed every time the files are updated. I have not seen this problem myself.

Unfortunately, I do not know the proper way to remove the account access from the NTFS filesystem. If you continue to have problems after your Linux update, I recommend one of the following options to reverse the procedure:

- Reformat the filesystem; this will destroy all data on it, but is the easiest option
- Request further instructions from someone more knowledgeable with Windows on how to remove the security entry for "Account Unknown"
- Open a bug or otherwise request further support from your Linux distribution provider, assuming your NTFS driver for Linux came with your Linux OS

Sorry I don't have a better solution for you myself at this time, but I'd love to hear your progress working through this.

**Comment by StuffOfSonny** [comment]
I have theory crafted a less damaging solution: backing up my steammapps folder, and nuking the old Steam folder, and making a new one.
I followed your guide, but I made the last minute decision to only apply the guide to the Steam folder. I can still run and execute files off of the rest of my Hard Drive just fine, but Steam's folders, files, and subfolders are in a tug of war.

It's possible that Steam will boot consistently, and I would just have to run the "Take Ownership" command on whatever Game folder I desire to play at the time. Thankfully WinAeroTweaker allows me to have the "Take Ownership" command in the context menu.
Still not as ideal as just letting two OSes share a drive, but this could just be an issue with NTFS, and I have been recommended  BRTFS on my next storage upgrade, whenever storage prices normalize again.

**Comment by FizzleFuze** [comment]
Sounds like a good idea. Thanks for following up. 🙂

I wonder if you would be interested in creating a separate partition for your steam library. This way you can mount it in Linux without mounting the rest of the drive. This is what I would do as a precaution as something isn't working correctly. 

Separate file systems and duplicate data are a bit of a pain but certainly the best choice for performance and security. I hope you find a drive at a good price soon. 👍

**Comment by jhgames1024** [comment]
I'm also having this problem, and after changing permissions so everyone can have full control to the Steam folder, it shows a fatal error "Failed to load steamui.dll"

**Comment by StuffOfSonny** [comment]
Yeah, I needed to take ownership of the Steam critical files for it to load.
My hotfix was to make a new folder that I would drag all of the loose files in the Steam Root into, run the Take Ownership command on it, and then drag the files back to the Root. You may also need to run Take Ownership on the bin folder too.

**Comment by FizzleFuze** [comment]
Sorry to hear this. Something is certainly not working correctly. 

Which version of Linux and Windows are you running?

**Comment by jhgames1024** [comment]
Ubuntu 20.04.2 LTS
Windows 10 Home Single Language 20H2 (Build 19042.1110)

**Comment by FizzleFuze** [comment]
Thank you!

It is very interesting to me that you both have reported the same problem on a debian (Ubuntu)-based kernel.

Does the workaround posted by StuffOfSonny above help with your problem?

I strongly encourage both of you to report this situation to your respective Linux distro bug system.

**Comment by jhgames1024** [comment]
I think it's something related to how this command works.

``sudo chown -R bob /mnt/Games/SteamLibrary``
`sudo chgrp -R bob /mnt/Games/SteamLibrary`

Since it only affects the specified path, not the entire partition.

**Comment by FizzleFuze** [comment]
Yes, I believe so I too. Those are basic Linux commands though so I can’t change them in any way. Someone who works on your distro will need to analyze how those commands hook in to the NTFS driver and what the driver is actually doing to the file system. ---- On Fri, 23 Jul 2021 09:58:19 -0400  ***@***.***> wrote ---- 
I think it's something related to how this command works. 
sudo chown -R bob /mnt/Games/SteamLibrary 
sudo chgrp -R bob /mnt/Games/SteamLibrary 
Since it only affects the specified path, not the entire partition.

**Comment by ioNelio** [comment]
I know this is late but I followed your tutorial on an arch based system and have got the same error. On my windows partition an "Account Unknown" has taken ownership of the steam library. I'd love to give more info but I really don't know what to look for when troubleshooting this problem. 

I guess I can try a different format that is shared between windows and linux if I'll get better results but I'd prefer to avoid that if possible.

**Comment by StuffOfSonny** [comment]
Minor follow up:
I haven't had the time/memory (brain wise, not Random Access) to boot back into Linux to try removing the permissions (part of me is waiting for SteamOS 3.0 to drop), but my workaround is as follows:
-I took my steamapps folder, and moved that to a different location
-Uninstalled Steam from my D:/
-Reinstalled Steam to my D:/
-Moved my steamapps back into the expected location

This allows Steam to boot and function as normal, but I still need to Take Ownership of each game whenever they install an update (Like Fallout 76 for example). Taking Ownership of singular games only takes a couple seconds, but running the command on the whole Steam folder would take nearly an hour.
This is made significantly more bearable by adding the Take Ownership command to the Context Menu via WinAero Tweaker.

As for "shared File Systems for Windows and Linux", the only other one I'm familiar with is BRTFS, which has a 3rd party driver for Windows, and an NTFS->BRTFS conversion tool. I've been warned to use it as a last ditch effort, however.

(Sorry for potential redundancies, it has been a while).

**Comment by mtricht** [comment]
Same issue here with Windows 11 and Manjaro. The README should have a warning about this since this is apparently *not* the more secure way as it breaks Steam on Windows.

**Comment by FizzleFuze** [comment]
One thing that comes to mind is that I have my game libraries separate from my Steam installation. 

In the guide, I wrote about changing the ownership on the Steam library, not the Steam installation itself. I don't expect Windows will be too happy when the ownership of the Steam .exe and .dlls are changed out from under it.

I'll update the guide soon to make this more clear.
It should be:
Windows owns steam
Linux owns (games) library

**Comment by StuffOfSonny** [comment]
At first, simply reinstalling Steam to Linux allowed me to see my library (though I kept flipflopping between the .deb version and the flatpack, for ease of use with Proton Glorious Eggroll), but that still resulted in the issue of Windows and Linux being unable to share my already installed Library. I don't have a lot of space to work with, so I wanted to avoid having second (or in some cases 3rd or more) installations of games. If this guide is unable, or uninterested in achieving such, then I will no longer have interest in this guide. Some day I would love to KVM my Windows (and possibly other OS) needs, and passing through shared drive access is a hard floor necessity to me.
